File: app/src/main/java/org/ole/planet/myplanet/callback/OnFilterListener.java
package org.ole.planet.myplanet.callback;

import java.util.List;
import java.util.Map;
import java.util.Set;

public interface OnFilterListener {
    public void filter(Set<String> subjects, Set<String> languages, Set<String> mediums, Set<String> levels);

    public Map<String, Set<String>> getData();

    public Map<String, Set<String>> getSelectedFilter();
}

File: app/src/main/java/org/ole/planet/myplanet/callback/OnFilterListener.kt
package org.ole.planet.myplanet.callback

interface OnFilterListener {
    fun filter(subjects: MutableSet<String>, languages: MutableSet<String>, mediums: MutableSet<String>, levels: MutableSet<String>): Int

    suspend fun getData(): Map<String, Set<String>>

    fun getSelectedFilter(): Map<String, Set<String>>

    fun getFilteredCount(subjects: Set<String>, languages: Set<String>, mediums: Set<String>, levels: Set<String>): Int

    fun clearAllFilters()
}
